Modules/_testcapimodule.c

```diff
@@ -2340,6 +2340,195 @@ get_timezone_utc_capi(PyObject* self, PyObject *args) {
+            hour, minute, second, microsecond,
+            Py_None,
+            PyDateTimeAPI->DateTimeType);
+    }
+    return rv;
+}
+
+static PyObject *
+get_datetime_fromdateandtimeandfold(PyObject *self, PyObject *args)
+{
+    PyObject *rv = NULL;
+    int year = 0, month = 0, day = 0;
+    int hour = 0, minute = 0, second = 0, microsecond = 0, fold = 0;
+    int macro = 0;
+
+    if (!PyArg_ParseTuple(
+        args, "iiiiiiii|p",
+        &year, &month, &day,
+        &hour, &minute, &second, &microsecond,
+        &fold,
+        &macro)) {
+
+        return NULL;
+    }
+
+    // Pass along to the API function
+    if (macro) {
+        rv = PyDateTime_FromDateAndTimeAndFold(
+            year, month, day,
+            hour, minute, second, microsecond,
+            fold);
+    }
+
+    else {
+        rv = PyDateTimeAPI->DateTime_FromDateAndTimeAndFold(
+            year, month, day,
+            hour, minute, second, microsecond,
+            Py_None,
+            fold,
+            PyDateTimeAPI->DateTimeType);
+    }
+
+        return rv;
+    }
+
+static PyObject *
+get_time_fromtime(PyObject *self, PyObject *args)
+{
+    PyObject *rv = NULL;
+    int hour = 0, minute = 0, second = 0, microsecond = 0;
+    int macro = 0;
+
+    if (!PyArg_ParseTuple(
+        args,
+        "iiii|p",
+        &hour, &minute, &second, &microsecond,
+        &macro)) {
+
+        return NULL;
+    }
+
+    // Pass along to the API function
+    if (macro) {
+        rv = PyTime_FromTime(hour, minute, second, microsecond);
+    }
+    else {
+        rv = PyDateTimeAPI->Time_FromTime(
+            hour, minute, second, microsecond,
+            Py_None,
+            PyDateTimeAPI->TimeType);
+    }
+
+    return rv;
+}
+
+static PyObject *
+get_time_fromtimeandfold(PyObject *self, PyObject *args)
+{
+    PyObject *rv = NULL;
+    int hour = 0, minute = 0, second = 0, microsecond = 0, fold = 0;
+    int macro = 0;
+
+    if (!PyArg_ParseTuple(
+        args,
+        "iiiii|p",
+        &hour, &minute, &second, &microsecond, &fold,
+        &macro)) {
+
+        return NULL;
+    }
+
+    // Pass along to the API function
+    if (macro) {
+        rv = PyTime_FromTimeAndFold(hour, minute, second, microsecond, fold);
+    }
+    else {
+        rv = PyDateTimeAPI->Time_FromTimeAndFold(
+            hour, minute, second, microsecond,
+            Py_None,
+            fold,
+            PyDateTimeAPI->TimeType);
+    }
+
+    return rv;
+}
+
+static PyObject *
+get_delta_fromdsu(PyObject *self, PyObject *args)
+{
+    PyObject *rv = NULL;
+    int days = 0, seconds = 0, microseconds = 0;
+    int macro = 0;
+
+    if (!PyArg_ParseTuple(
+        args,
+        "iii|p",
+        &days, &seconds, &microseconds,
+        &macro)) {
+
+        return NULL;
+    }
+
+    // Pass along to the API function
+    if (macro) {
+        rv = PyDelta_FromDSU(days, seconds, microseconds);
+    }
+    else {
+        rv = PyDateTimeAPI->Delta_FromDelta(
```

Comment:
Breaking long lines: if you can, break after commas in the outermost argument list. Always indent continuation lines appropriately, e.g.: